library/src/main/kotlin/com/livefront/gsonkotlinadapter/util/KParameter.kt

```diff
@@ -3,8 +3,28 @@ package com.livefront.gsonkotlinadapter.util
 import com.google.gson.annotations.SerializedName
 import java.lang.reflect.Field
 import java.lang.reflect.Modifier
+import kotlin.reflect.KClass
 import kotlin.reflect.KParameter
 
+/**
+ * Returns the default value of this [KParameter] or `null`.
```

Comment:
We should explicitly call out how this is not the same as the default value supplied to a function or constructor call:

```kotlin
/*
 * ...
 * Note that this refers to a constant type-specific default value (i.e. null for objects, 0 for numeric primitives, etc.) as
 * opposed to manually-supplied defaults found in a function / constructor parameter list.
 */
```